Cython/Compiler/ModuleNode.py

```diff
@@ -3414,6 +3414,15 @@ def generate_module_cleanup_func(self, env, code):
                    Naming.cleanup_cname)
         code.enter_cfunc_scope(env)
 
+        # TODO - this should go away when module-state has been refactored more and
+        # we are able to access the module state through "self". Currently the
+        # `#define` for each entry forces us to access it through PyState_FindModule
+        # which is sometime unreliable during destruction
+        # (e.g. during interpreter shutdown).
+        # In that case the safest thing is to give up.
+        code.putln(f"if (!PyState_FindModule(&{Naming.pymoduledef_cname}))")
+        code.putln("return;")
```

Comment:
I think this change won't be needed after https://github.com/cython/cython/pull/5323. That's the "right" fix, and this is just a fairly quick temporary measure.

Reply:
I'm less convinced it'll be completely unneeded after a bit more thought (since destructors can call into arbitrary code outside the module's cleanup function). Further refactors to reduce the dependence of `PyState_FindModule` will help though. And we might be able to split it into cached constants (safe) and everything else (unsafe).